lib/credit_card_widget.dart

```diff
@@ -185,15 +229,38 @@ class _CreditCardWidgetState extends State<CreditCardWidget>
       vsync: this,
     );
 
+    controller.addStatusListener((AnimationStatus status) {
+      if (status == AnimationStatus.forward) {
+        isAnimating = true;
+      } else if (status == AnimationStatus.reverse) {
+        isAnimating = true;
+      } else if (status == AnimationStatus.completed) {
+        isAnimating = false;
+      } else {
+        isAnimating = false;
+      }
+    });
+
+    _initializeFloatingAnimation();
     _gradientSetup();
     _updateRotations(false);
   }
 
+  @override
+  void didChangeDependencies() {
+    orientation = MediaQuery.of(context).orientation;
```

Comment:
can we write this code like this ?
`orientation ??= MediaQuery.of(context).orientation;`

Reply:
I believe it would be better to always use the updated values of the `MediaQuery` object.